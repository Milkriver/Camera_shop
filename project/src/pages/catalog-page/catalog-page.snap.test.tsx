import {render} from '@testing-library/react';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import HistoryRouter from '../../components/history-router/history-router';
import CatalogPage from './catalog-page';


describe('Component: CatalogPage', () => {
  test('should render correctly', () => {
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);

    const store = mockStore({
      OFFERS: {
        offers: [],
      },
      DATA: {
        activePaginationPage: 1,
      }
    });

    const history = createMemoryHistory();

    const {container} = render(
      <Provider store={store}>
        <HistoryRouter history={history}>
          <CatalogPage />
        </HistoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
