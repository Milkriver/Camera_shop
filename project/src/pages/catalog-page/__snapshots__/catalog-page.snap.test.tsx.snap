// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CatalogPage should render correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    <header
      class="header"
      id="header"
    >
      <div
        class="container"
      >
        <a
          aria-label="Переход на главную"
          class="header__logo"
          href="/"
        >
          <img
            alt="logo"
            height="36"
            src="/img/sprite/icon-logo.svg"
            width="100"
          />
        </a>
        <nav
          class="main-nav header__main-nav"
        >
          <ul
            class="main-nav__list"
          >
            <li
              class="main-nav__item"
            >
              <a
                class="main-nav__link"
                href="/catalog.html"
              >
                Каталог
              </a>
            </li>
            <li
              class="main-nav__item"
            >
              <a
                class="main-nav__link"
                href="/"
              >
                Гарантии
              </a>
            </li>
            <li
              class="main-nav__item"
            >
              <a
                class="main-nav__link"
                href="/"
              >
                Доставка
              </a>
            </li>
            <li
              class="main-nav__item"
            >
              <a
                class="main-nav__link"
                href="/"
              >
                О компании
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="form-search "
        >
          <form>
            <label>
              <svg
                aria-hidden="true"
                class="form-search__icon"
                height="16"
                width="16"
              >
                <use
                  xlink:href="#icon-lens"
                />
              </svg>
              <input
                autocomplete="off"
                class="form-search__input"
                placeholder="Поиск по сайту"
                type="text"
                value=""
              />
            </label>
            <ul
              class="form-search__select-list"
            />
          </form>
          <button
            class="form-search__reset"
            type="reset"
          >
            <svg
              aria-hidden="true"
              height="10"
              width="10"
            >
              <use
                xlink:href="#icon-close"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              Сбросить поиск
            </span>
          </button>
        </div>
        <a
          class="header__basket-link"
          href="/"
        >
          <svg
            aria-hidden="true"
            height="16"
            width="16"
          >
            <use
              xlink:href="#icon-basket"
            />
          </svg>
        </a>
      </div>
    </header>
    <main>
      <div
        class="banner"
      />
      <div
        class="page-content"
      >
        <div
          class="breadcrumbs"
        >
          <div
            class="container"
          >
            <ul
              class="breadcrumbs__list"
            >
              <li
                class="breadcrumbs__item"
              >
                <a
                  class="breadcrumbs__link"
                  href="/"
                >
                  Главная
                  <svg
                    aria-hidden="true"
                    height="8"
                    width="5"
                  >
                    <use
                      xlink:href="#icon-arrow-mini"
                    />
                  </svg>
                </a>
              </li>
            </ul>
          </div>
        </div>
        <section
          class="catalog"
        >
          <div
            class="container"
          >
            <h1
              class="title title--h2"
            >
              Каталог фото- и видеотехники
            </h1>
            <div
              class="page-content__columns"
            >
              <div
                class="catalog__aside"
              >
                <div
                  class="catalog-filter"
                >
                  <form
                    action="#"
                  >
                    <h2
                      class="visually-hidden"
                    >
                      Фильтр
                    </h2>
                    <fieldset
                      class="catalog-filter__block"
                    >
                      <legend
                        class="title title--h5"
                      >
                        Цена, ₽
                      </legend>
                      <div
                        class="catalog-filter__price-range"
                      >
                        <div
                          class="custom-input"
                        >
                          <label>
                            <input
                              name="price"
                              placeholder="0"
                              type="number"
                              value=""
                            />
                          </label>
                        </div>
                        <div
                          class="custom-input"
                        >
                          <label>
                            <input
                              name="priceUp"
                              placeholder="0"
                              type="number"
                              value=""
                            />
                          </label>
                        </div>
                      </div>
                    </fieldset>
                    <fieldset
                      class="catalog-filter__block"
                    >
                      <legend
                        class="title title--h5"
                      >
                        Категория
                      </legend>
                      <div
                        class="custom-checkbox catalog-filter__item"
                      >
                        <label>
                          <input
                            name="photocamera"
                            type="checkbox"
                          />
                          <span
                            class="custom-checkbox__icon"
                          />
                          <span
                            class="custom-checkbox__label"
                          >
                            Фотокамера
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-checkbox catalog-filter__item"
                      >
                        <label>
                          <input
                            name="videocamera"
                            type="checkbox"
                          />
                          <span
                            class="custom-checkbox__icon"
                          />
                          <span
                            class="custom-checkbox__label"
                          >
                            Видеокамера
                          </span>
                        </label>
                      </div>
                    </fieldset>
                    <fieldset
                      class="catalog-filter__block"
                    >
                      <legend
                        class="title title--h5"
                      >
                        Тип камеры
                      </legend>
                      <div
                        class="custom-checkbox catalog-filter__item"
                      >
                        <label>
                          <input
                            name="digital"
                            type="checkbox"
                          />
                          <span
                            class="custom-checkbox__icon"
                          />
                          <span
                            class="custom-checkbox__label"
                          >
                            Цифровая
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-checkbox catalog-filter__item"
                      >
                        <label>
                          <input
                            name="film"
                            type="checkbox"
                          />
                          <span
                            class="custom-checkbox__icon"
                          />
                          <span
                            class="custom-checkbox__label"
                          >
                            Плёночная
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-checkbox catalog-filter__item"
                      >
                        <label>
                          <input
                            name="snapshot"
                            type="checkbox"
                          />
                          <span
                            class="custom-checkbox__icon"
                          />
                          <span
                            class="custom-checkbox__label"
                          >
                            Моментальная
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-checkbox catalog-filter__item"
                      >
                        <label>
                          <input
                            name="collection"
                            type="checkbox"
                          />
                          <span
                            class="custom-checkbox__icon"
                          />
                          <span
                            class="custom-checkbox__label"
                          >
                            Коллекционная
                          </span>
                        </label>
                      </div>
                    </fieldset>
                    <fieldset
                      class="catalog-filter__block"
                    >
                      <legend
                        class="title title--h5"
                      >
                        Уровень
                      </legend>
                      <div
                        class="custom-checkbox catalog-filter__item"
                      >
                        <label>
                          <input
                            name="zero"
                            type="checkbox"
                          />
                          <span
                            class="custom-checkbox__icon"
                          />
                          <span
                            class="custom-checkbox__label"
                          >
                            Нулевой
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-checkbox catalog-filter__item"
                      >
                        <label>
                          <input
                            name="non-professional"
                            type="checkbox"
                          />
                          <span
                            class="custom-checkbox__icon"
                          />
                          <span
                            class="custom-checkbox__label"
                          >
                            Любительский
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-checkbox catalog-filter__item"
                      >
                        <label>
                          <input
                            name="professional"
                            type="checkbox"
                          />
                          <span
                            class="custom-checkbox__icon"
                          />
                          <span
                            class="custom-checkbox__label"
                          >
                            Профессиональный
                          </span>
                        </label>
                      </div>
                    </fieldset>
                    <button
                      class="btn catalog-filter__reset-btn"
                      type="reset"
                    >
                      Сбросить фильтры
                    </button>
                  </form>
                </div>
              </div>
              <div
                class="catalog__content"
              >
                <div
                  class="catalog-sort"
                >
                  <form
                    action="#"
                  >
                    <div
                      class="catalog-sort__inner"
                    >
                      <p
                        class="title title--h5"
                      >
                        Сортировать:
                      </p>
                      <div
                        class="catalog-sort__type"
                      >
                        <div
                          class="catalog-sort__btn-text"
                        >
                          <input
                            id="price"
                            name="sort"
                            type="radio"
                            value="price"
                          />
                          <label
                            for="price"
                          >
                            по цене
                          </label>
                        </div>
                        <div
                          class="catalog-sort__btn-text"
                        >
                          <input
                            id="rating"
                            name="sort"
                            type="radio"
                            value="rating"
                          />
                          <label
                            for="rating"
                          >
                            по популярности
                          </label>
                        </div>
                      </div>
                      <div
                        class="catalog-sort__order"
                      >
                        <div
                          class="catalog-sort__btn catalog-sort__btn--up"
                        >
                          <input
                            aria-label="по возрастанию"
                            id="up"
                            name="sort-icon"
                            type="radio"
                            value="asc"
                          />
                          <label
                            for="up"
                          >
                            <svg
                              aria-hidden="true"
                              height="14"
                              width="16"
                            >
                              <use
                                xlink:href="#icon-sort"
                              />
                            </svg>
                          </label>
                        </div>
                        <div
                          class="catalog-sort__btn catalog-sort__btn--down"
                        >
                          <input
                            aria-label="по убыванию"
                            id="down"
                            name="sort-icon"
                            type="radio"
                            value="desc"
                          />
                          <label
                            for="down"
                          >
                            <svg
                              aria-hidden="true"
                              height="14"
                              width="16"
                            >
                              <use
                                xlink:href="#icon-sort"
                              />
                            </svg>
                          </label>
                        </div>
                      </div>
                    </div>
                  </form>
                </div>
                <div
                  class="cards catalog__cards"
                >
                  <div>
                    По вашему запросу ничего не найдено
                  </div>
                </div>
                <div
                  class="pagination"
                >
                  <ul
                    class="pagination__list"
                  />
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </main>
    <footer
      class="footer"
    >
      <div
        class="container"
      >
        <div
          class="footer__info"
        >
          <a
            aria-label="Переход на главную"
            class="footer__logo"
            href="index.html"
          >
            <svg
              aria-hidden="true"
              height="36"
              width="100"
            >
              <use
                xlink:href="#icon-logo-mono"
              />
            </svg>
          </a>
          <p
            class="footer__description"
          >
            Интернет-магазин фото- и видеотехники
          </p>
          <ul
            class="social"
          >
            <li
              class="social__item"
            >
              <a
                aria-label="Переход на страницу вконтатке"
                class="link"
                href="/"
              >
                <svg
                  aria-hidden="true"
                  height="20"
                  width="20"
                >
                  <use
                    xlink:href="#icon-vk"
                  />
                </svg>
              </a>
            </li>
            <li
              class="social__item"
            >
              <a
                aria-label="Переход на страницу pinterest"
                class="link"
                href="/"
              >
                <svg
                  aria-hidden="true"
                  height="20"
                  width="20"
                >
                  <use
                    xlink:href="#icon-pinterest"
                  />
                </svg>
              </a>
            </li>
            <li
              class="social__item"
            >
              <a
                aria-label="Переход на страницу reddit"
                class="link"
                href="/"
              >
                <svg
                  aria-hidden="true"
                  height="20"
                  width="20"
                >
                  <use
                    xlink:href="#icon-reddit"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
        <ul
          class="footer__nav"
        >
          <li
            class="footer__nav-item"
          >
            <p
              class="footer__title"
            >
              Навигация
            </p>
            <ul
              class="footer__list"
            >
              <li
                class="footer__item"
              >
                <a
                  class="link"
                  href="/catalog.html"
                >
                  Каталог
                </a>
              </li>
              <li
                class="footer__item"
              >
                <a
                  class="link"
                  href="/"
                >
                  Гарантии
                </a>
              </li>
              <li
                class="footer__item"
              >
                <a
                  class="link"
                  href="/"
                >
                  Доставка
                </a>
              </li>
              <li
                class="footer__item"
              >
                <a
                  class="link"
                  href="/"
                >
                  О компании
                </a>
              </li>
            </ul>
          </li>
          <li
            class="footer__nav-item"
          >
            <p
              class="footer__title"
            >
              Ресурсы
            </p>
            <ul
              class="footer__list"
            >
              <li
                class="footer__item"
              >
                <a
                  class="link"
                  href="/"
                >
                  Курсы операторов
                </a>
              </li>
              <li
                class="footer__item"
              >
                <a
                  class="link"
                  href="/"
                >
                  Блог
                </a>
              </li>
              <li
                class="footer__item"
              >
                <a
                  class="link"
                  href="/"
                >
                  Сообщество
                </a>
              </li>
            </ul>
          </li>
          <li
            class="footer__nav-item"
          >
            <p
              class="footer__title"
            >
              Поддержка
            </p>
            <ul
              class="footer__list"
            >
              <li
                class="footer__item"
              >
                <a
                  class="link"
                  href="/"
                >
                  FAQ
                </a>
              </li>
              <li
                class="footer__item"
              >
                <a
                  class="link"
                  href="/"
                >
                  Задать вопрос
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </footer>
  </div>
</div>
`;
